### PR TITLE
feat: implement PUT /api/events validation and client interface for I…

### PR DESCRIPTION
**Pull request by yhua045**
…ssue #2

Implements comprehensive validation and API client method for updating roster assignments:

- Added EventNotFoundError exception for 404 responses
- Implemented RosterAPIClient.update_roster_assignment() method with:
  * Validation for service_info_id, date, role, name, and category
  * Support for both minimal parameters and full serviceInfo object
  * Whitespace normalization for names and roles
  * Comprehensive error handling (404, 400, 500, network errors)
  * Audit logging for updates
- Added sanitize_name() validator utility
- Added 16 unit tests covering validation, error handling, and edge cases
- All 46 tests passing

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/utils/validators.py [hunk 1]
```diff
@@ -67,4 +67,23 @@ def normalize_category(category: str) -> str:
...
+
+    Returns:
+        Sanitized name with normalized whitespace
+    """
+    if not name:
+        return ""
+
+    # Normalize whitespace: strip leading/trailing and collapse internal whitespace
+    import re
+    normalized = re.sub(r'\s+', ' ', name.strip())
```
[the remark this answers is not in the document]
Comment (yhua045) [reply src/utils/validators.py:88]: @copilot, please go ahead and make the suggest changes
